### PR TITLE
fix: create auxclasspath file for each spotbugs task

### PR DESCRIPTION
**Pull request by OdysseusLives**
and auxclasspath file is removed and created instead of truncated in place

This addresses #259 by avoiding re-creating the same file in any invocation, and adds some additional information if a failure occurs.

### CODE REVIEW

File: src/main/groovy/com/github/spotbugs/snom/internal/SpotBugsRunner.java [hunk 1]
```diff
@@ -127,14 +127,23 @@ private String createFileForAuxClasspath(SpotBugsTask task) {
           Paths.get(
               task.getProject().getBuildDir().getAbsolutePath(),
               "spotbugs",
-              "spotbugs-auxclasspath");
-      Files.createDirectories(auxClasspathFile.getParent());
-      Files.createFile(auxClasspathFile);
-      Files.write(auxClasspathFile, auxClasspath.getBytes(), StandardOpenOption.TRUNCATE_EXISTING);
-      return auxClasspathFile.normalize().toString();
+              "spotbugs-auxclasspath-for-" + task.getName());
```
Comment (KengoTODA) [inline src/main/groovy/com/github/spotbugs/snom/internal/SpotBugsRunner.java:130]: just feeling: `"auxclasspath", task.getName());` would be good replacement, then the directory hierarchy will be more intuitive like:

```
project/
  - build/
    - spotbugs/
      - auxclasspath/
        - spotbugsMain
        - spotbugsTest
```

File: src/main/groovy/com/github/spotbugs/snom/internal/SpotBugsRunner.java [hunk 2]
```diff
@@ -127,14 +127,23 @@ private String createFileForAuxClasspath(SpotBugsTask task) {
...
+      try {
+        Files.createDirectories(auxClasspathFile.getParent());
+        if (Files.exists(auxClasspathFile)) {
+          Files.delete(auxClasspathFile);
+        }
+        Files.createFile(auxClasspathFile);
+        Files.write(auxClasspathFile, auxClasspath.getBytes(), StandardOpenOption.WRITE);
```
Comment (KengoTODA) [inline src/main/groovy/com/github/spotbugs/snom/internal/SpotBugsRunner.java:137]: What is our motivation to replace `TRUNCATE_EXISTING` with `WRITE`?
and I guess `TRUNCATE_EXISTING` lets us remove the lines 133-136? (sorry I didn't noticed it previously)

File: src/main/groovy/com/github/spotbugs/snom/internal/SpotBugsRunner.java [hunk 3]
```diff
@@ -127,14 +127,23 @@ private String createFileForAuxClasspath(SpotBugsTask task) {
...
+      } catch (Exception e) {
+        throw new GradleException(
+            "Could not create auxiliary classpath file for SpotBugsTask at "
+                + auxClasspathFile.normalize().toString(),
+            e);
+      }
     } catch (Exception e) {
-      // oops
-      throw new GradleException("Could not create auxiliary classpath file for SpotBugsTask");
+      throw new GradleException("Could not create auxiliary classpath file for SpotBugsTask", e);
```
Comment (KengoTODA) [inline src/main/groovy/com/github/spotbugs/snom/internal/SpotBugsRunner.java:146]: 👍

File: src/main/groovy/com/github/spotbugs/snom/internal/SpotBugsRunner.java (hunk 2 above)
Comment (OdysseusLives) [reply src/main/groovy/com/github/spotbugs/snom/internal/SpotBugsRunner.java:137]: The motivation was to delete and create the file each time, but the outcome should be taken care of with the directory hierarchy however.

File: src/main/groovy/com/github/spotbugs/snom/internal/SpotBugsRunner.java (hunk 1 above)
Comment (OdysseusLives) [reply src/main/groovy/com/github/spotbugs/snom/internal/SpotBugsRunner.java]: I like that 👍

**Pull request merged**